staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/dispatcher.go

```diff
@@ -62,30 +62,55 @@ func newValidatingDispatcher(p *Plugin) func(cm *webhookutil.ClientManager) gene
 	}
 }
 
+var _ generic.VersionedAttributeAccessor = &versionedAttributeAccessor{}
+
+type versionedAttributeAccessor struct {
+	versionedAttrs map[schema.GroupVersionKind]*admission.VersionedAttributes
+}
+
+func (v *versionedAttributeAccessor) VersionedAttribute(attr admission.Attributes, o admission.ObjectInterfaces, gvk schema.GroupVersionKind) (*admission.VersionedAttributes, error) {
```

Comment:
I think this interface can be trimmed to

```go
VersionedAttribute(gvk schema.GroupVersionKind) (*admission.VersionedAttributes, error)
```

and `attr admission.Attributes, o admission.ObjectInterfaces` baked into the accessor (they're the same on all invocations)

Then we don't need a separate VersionedAttributeFromCache implementation

Reply:
done! this is much cleaner ty